cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppFabricHttpHandler.java

```diff
@@ -756,9 +773,62 @@ public void sparkSpecification(HttpRequest request, HttpResponder responder,
   public void workflowSpecification(HttpRequest request, HttpResponder responder,
                                     @PathParam("app-id") final String appId,
                                     @PathParam("workflow-id")final String workflowId) {
-    programLifecycleHttpHandler.programSpecification(RESTMigrationUtils.rewriteV2RequestToV3(request), responder,
```

Comment:
We changed WorkflowSpecifications in 2.8. However we still want V2 WorkflowSpecification api to return the data in the format which is compatible to 2.7, which is why special handling is needed here.